model/src/main/kotlin/licenses/ResolvedLicenseInfo.kt

```diff
@@ -65,6 +66,23 @@ data class ResolvedLicenseInfo(
 ) : Iterable<ResolvedLicense> by licenses {
     operator fun get(license: SpdxSingleLicenseExpression): ResolvedLicense? = find { it.license == license }
 
+    /**
```

Comment:
IIUC this resolved license info class has all configuration relevant for computing the needed information in its attributes. So, what's the reason for not adding an attribute for the applicable license choices to the data model?

For `ResolvedLicenseLocation` there is `appliedCuration` and `matchingPathExcludes` there is the curations and path excludes already applied, and additionally there is the properties for what was applied. The analog thing for choices would be that choices are already applied and there is an `applied choices` property?

Note: I'm haven't myself yet fully understood this area of the code, so I rather want to get the discussion going, but don't intent to say it's wrong as-is.

Reply:
Yes we thought about this.
But the license choice should only be applied to the licenses used in the evaluator and the reporter, and therefore make use of the specified `LicenseView`s.

This implementation decision was a direct result of a discussion in the developers meeting on 2021-03-04, we developed it in pair programming with @mnonnenmacher.

Reply:
Exactly, the problem is that the license choice can only be applied if the sub-expression matches, and if it does can depend on the used `LicenseView`. So the available license choices could be provided by the `LicenseInfo` as mentioned in another comment, but they can only be applied in combination with a `LicenseView`, for example in the evaluator rules.